### ISSUE TITLE
UDM Pro has not reported statistics for more than 10m

### ISSUE THREAD

**Issue by fjh40** [patricegautier/unifiZabbix#13]
Hi there Patrice,

First of all, thank you for developing this script, the integration in Zabbix and documenting the process.
After a lot of troubleshooting I have finally connected my Zabbix and my Ubiquiti Dream Machine Pro using SSH via a generated key. 
The only issue that I'm currently facing is that Zabbix is not displaying all of the data from the UDM Pro, would you happen to know by any chance if there's anything I did wrong or can check to fix this?

In the logs it gives me a message about not being able to read data from the SSH server, however some data is being received, like load average.
Also I am able to connect using SSH to the UDM Pro and retrieving data using the JSON string command.

I have included the logs, screenshots of my configuration and tests in the following imgur link with descriptions to keep this post on Github neat:
https://imgur.com/a/UQCFhME

Please let me know if you require any additional information

Kind regards,
Felix

**Comment by patricegautier** [comment]
mm.. if you go to the configuration screen for the UDMP in zabbix and look at the items there, and especially the ones that show as 'not supported', is there any useful error message on those?

**Comment by patricegautier** [comment]
also, anything useful in /var/log/messages on the UDMP?

**Comment by fjh40** [comment]
Thanks for the reply! I've looked at the error messages it gives me in Zabbix under 'items' and also on the /var/log/messages on my UDMP. The SSH session seems to be initiated correctly, but then closed almost immediately as well as far as I can tell. Also a few errors it gives in Zabbix is not being able to find the script? I'll try and look into that as well, maybe some sort of permission problem on the MCA script on my server.

Please see the Imgur link once again for additional screenshots:
https://imgur.com/a/un7meDU

Thank you in advance for the support!

**Comment by fjh40** [comment]
Okay so I've double checked the script permissions, location and it being included in the config as well.
I've also deleted and added my UDMP host again, the sensors go to 'supported' for a half a minute or something but then again go to status 'not supported'. 

Here is my Zabbix server log from /var/log/zabbix/zabbix_server.log (Not being able to read DHCP is correct, since I use Pi-hole for that)
![image](https://user-images.githubusercontent.com/33373570/116479432-eb7bbc80-a87f-11eb-998a-84af04d9a58e.png)

And here is my UDMP log, once again it seems that the SSH session is initiated but then immediately closed:
![image](https://user-images.githubusercontent.com/33373570/116479502-09492180-a880-11eb-808d-6c61d81a7526.png)

My UDMP is running firmware version 1.9.2, your script should be doing it's work fine since I can still get a lot of data in a JSON format when running the following command: sudo -u zabbix /usr/lib/zabbix/externalscripts/mca-dump-short.sh -d 10.20.20.1 -u root -i /home/felix/.ssh/zb_id_rsa -t UDMP

For now I'm out of ideas and might have to give up on this project for now. Maybe I'll get some ideas in the upcomings weeks where to look!

**Comment by fjh40** [comment]
Excuse me for the amount of posts, however it's working! (Almost fully)
About 15 minutes ago I noticed that I had forgot to uncomment the include path for the ExternalScripts folder in the Zabbix config. At first I thought that was it, but after readding my UDMP as a host it still didn't work, now it suddenly does? :)

The only thing which is still not supported is the 'uptime', but I'm not sure if this is a restriction on the UDMP side, your script or my Zabbix configuration.

![image](https://user-images.githubusercontent.com/33373570/116480583-ddc73680-a881-11eb-83ec-0057820d4d62.png)

![image](https://user-images.githubusercontent.com/33373570/116480708-21ba3b80-a882-11eb-8675-0e22a1030bdb.png)

The only thing which is a bit weird is the CPU usage which indicates 50%, but I feel like that might be a single core which is under load, since the UDMP indicates it's CPU is sitting around 18% (it has a quad core CPU). However it does seem like the rest of the stats are all good! And I believe I've also set the polling interval on Zabbix to a minute, so then again the CPU load might have been higher at that specific measurement.

Now I just have to figure out how to make beautiful graphs, I think for another project I'll probably connect Grafana to Zabbix for nicer graphs, but that's a project for another day when I've mastered Zabbix and all my devices are working!

Oh and if you might know why my uptime sensor isn't working please let me know, but don't spend too much time on it, it's not that big of a deal breaker. However I would like to clear all the error messages ^^

Kind regards,
Felix

**Comment by patricegautier** [comment]
Glad to hear it's working, sad it was magic!

Re Uptime:  does the uptime item in the host config have any error message?  it's a separate SSH invocation right now..

Re: graphs:  I ended building the graphs in Zabbix because I didn't want to have yet another tool to maintain, but Grafana is certainly very popular..

Re: CPU usage: I am getting a good match between the CPU usage in Zabbix and the one reported in the Unifi controller in general.  That value can vary pretty fast depending on what is going on, so that could be the source of the discrepancy.

**Comment by fjh40** [comment]
Hi Patrice,

I've waited a few days before responding to see how everything is working out, so far I must say I am very pleased with this setup! Everything has been working flawless so far, and even the uptime has started working out of nowhere as well... Although I'm not quite sure why this unit is in 'ms' and the graph keeps ascending. Maybe I'm missing something here? I guess it gets the job done tho, assuming the graph starts from 0 again at a reboot. (See attached screenshot).

Anyways, it seems that I had some misconfiguration at the start (also because I just started learning Zabbix and wasn't quite sure what I was doing). The limited available documentation on this script was a bit of a challenge the first few hours, but I've managed to get it all working nicely!

Also I understand that you don't 'force' Grafana, although it is one of the best in my opinion for showing graphs it still is another install / setup to perform. 

Thank you so much for the assistance and the awesome software!

Kind regards,
Felix

![Screenshot from 2021-05-02 01-34-38](https://user-images.githubusercontent.com/33373570/116797496-97bedc80-aae6-11eb-9da8-83a2b96a3bba.png)

**Comment by patricegautier** [comment]
Excellent!  I might get inspired to look at Grafana..

**Comment by fjh40** [comment]
Haha I can definately recommend Grafana, very awesome looking software. Once you get the hang of it I'm pretty sure you'll never go back :)

Also I forgot to include in my last post that the high CPU usage is indeed correct, it seems that the UDM Pro's 'settings' page is not showing the usage correctly. When I ran a 'top' command it indeed showed the same CPU usage as Zabbix (which makes sense, since it reads it from SSH as well). Also when enabling the old Web UI from Ubiquiti the CPU usage is shown correctly as well, so it seems I got bamboozled by the new incorrect Web UI from the UDM Pro.

Have a nice day!
